### DISCUSSION TITLE
Support for PlainText without a Schema

### DISCUSSION THREAD

**Discussion by Perry45** [sngular/kloadgen#286] [Ideas]
Hello,

I just wanted to ask if there are any plans to support also PlainText without a Schema like PepperBox. KLoadGen is really great, but I have Topics with a Avro Schema and Topics without a Schema where I just want to utilize the org.apache.kafka.common.serialization.StringSerializer. In PepperBox I can just enter a PlainText (as json) and send it to the topic. So at the moment I have to include both tools for my tests.

It would be amazing if KLoadGen would also support that as well. Or did you intentionally choose not to support that?

**Comment by IsaUlloa** [comment]
Hi @Perry45 ,

We have decided that we will support PlainText, but we will not support the generation of Random values from a text file. We will process the file and serialize the content to Kafka with org.apache.kafka.common.serialization.StringSerializer.

We will open an issue #287 from this discussion and we will update our news over there.

Thank you for rising this suggestion

**Reply by Perry45** [reply]
Ok great. I don't need the Random values. I would still be able to use JMeter Variables in the PlainText, am I?

**Reply by IsaUlloa** [reply]
Hi @Perry45 ,

Once we had added the support for PlainText, you will be able to use JMeter Variables with the function '__eval'. You can check it out here: https://jmeter.apache.org/usermanual/functions.html#__eval

Cheers
